airgun/utils.py

```diff
@@ -69,3 +73,22 @@ def get_widget_by_name(widget_root, widget_name):
                 )
         widget = getattr(widget, name)
     return widget
+
+
+def retry_navigation(method):
+    """Decorator to invoke method one or more times, if TimedOutError is raised."""
+
+    @functools.wraps(method)
+    def retry_wrapper(*args, **kwargs):
+        attempts = 3
+        for i in range(attempts):
+            try:
+                return method(*args, **kwargs)
+            except (Exception):
```

Comment:
https://github.com/SatelliteQE/airgun/pull/643/files#r777978323

Reply:
OR you can simply list the exceptions e.g (TimeoutError, NavigationTriesExceeded, etc) on when to retry!

Reply:
I see u opened a cherrypick of same issue in 7.0.